libs/blocks/merch/merch.js

```diff
@@ -198,22 +198,41 @@ export async function fetchCheckoutLinkConfigs(base = '') {
   return fetchCheckoutLinkConfigs.promise;
 }
 
-export async function getCheckoutLinkConfig(productFamily) {
+/**
+ * @param {*} productFamily product family code
+ * @param {*} productCode product code, can be undefined
+ * @param {*} paCode product arrangement code, can be undefined
+ * @returns checkout link config for the given product family and code
+ */
+export async function getCheckoutLinkConfig(productFamily, productCode, paCode) {
   let { base } = getConfig();
   if (/\.page$/.test(document.location.origin)) {
     /* c8 ignore next 2 */
     base = base.replace('.live', '.page');
   }
   const checkoutLinkConfigs = await fetchCheckoutLinkConfigs(base);
   const { locale: { region } } = getConfig();
-  const productFamilyConfigs = checkoutLinkConfigs.data?.filter(
-    ({ [NAME_PRODUCT_FAMILY]: mappingProductFamily }) => mappingProductFamily === productFamily,
-  );
-  if (productFamilyConfigs.length === 0) return undefined;
-  const checkoutLinkConfig = productFamilyConfigs.find(
+  let productCheckoutLinkConfigs = [];
+  if (paCode) {
+    productCheckoutLinkConfigs = checkoutLinkConfigs.data?.filter(
```

Comment:
i would rather only use the filter function once, with 3 tests in there, no? only one traversal of the array, with quick test looks quicker to me

Reply:
no, in that case, it will likely find first the entry with the product family rather than pa code  unless we require GWP to author product code and pa codes entries first.

Reply:
yes, but typical case with a family code line match implies traversal and test of 100 entries (l217) (no match), then another traversal of 100 entries (l221) (no match), then finally traversal of 100 entries to filter the right ones in l227.
my proposal was to do one traversal instead

Reply:
I think I can do it with a reducer logic, let me give it a try.

Reply:
I noticed that the column value is either family or paCode or productCode, is it safe?
shouldn't it be 
family_productCode_productArrangementCode?
where productCode and paCode are optional?

Reply:
@3ch023 I'll update the column name and the code after https://jira.corp.adobe.com/browse/ODM-19213 is done.